src/solver/variable/economy/STStorageCashFlowByCluster.h

```diff
@@ -0,0 +1,305 @@
+
+            for (unsigned int i = 0; i != nbClusters_; ++i)
+            {
+                AncestorType::pResults[i].initializeFromStudy(*study);
+                AncestorType::pResults[i].reset();
+            }
+        }
+        else
+        {
+            for (unsigned int numSpace = 0; numSpace < pNbYearsParallel; numSpace++)
+            {
+                pValuesForTheCurrentYear[numSpace] = nullptr;
+            }
+
+            AncestorType::pResults.clear();
+        }
+        // Next
+        NextType::initializeFromArea(study, area);
+    }
+
+    size_t getMaxNumberColumns() const
+    {
+        return nbClusters_ * ResultsType::count;
+    }
+
+    void yearBegin(unsigned int year, unsigned int numSpace)
+    {
+        // Reset the values for the current year
+        for (unsigned int i = 0; i != nbClusters_; ++i)
+        {
+            pValuesForTheCurrentYear[numSpace][i].reset();
+        }
+        // Next variable
+        NextType::yearBegin(year, numSpace);
+    }
+
+    void yearEnd(unsigned int year, unsigned int numSpace)
+    {
+        for (unsigned int clusterIndex = 0; clusterIndex < nbClusters_; ++clusterIndex)
+        {
+            // Compute all statistics from hourly results for the current year (daily, weekly, monthly, ...)
+            pValuesForTheCurrentYear[numSpace][clusterIndex].computeStatisticsForTheCurrentYear();
+        }
+        // Next variable
+        NextType::yearEnd(year, numSpace);
+    }
+
+    void computeSummary(std::map<unsigned int, unsigned int>& numSpaceToYear,
+                        unsigned int nbYearsForCurrentSummary)
+    {
+        for (unsigned int numSpace = 0; numSpace < nbYearsForCurrentSummary; ++numSpace)
+        {
+            for (unsigned int clusterIndex = 0; clusterIndex < nbClusters_; ++clusterIndex)
+            {
+                // Merge all those values with the global results
+                AncestorType::pResults[clusterIndex].merge(numSpaceToYear[numSpace],
+                                                pValuesForTheCurrentYear[numSpace][clusterIndex]);
+            }
+        }
+
+        // Next variable
+        NextType::computeSummary(numSpaceToYear, nbYearsForCurrentSummary);
+    }
+
+    void hourBegin(unsigned int hourInTheYear)
+    {
+        // Next variable
+        NextType::hourBegin(hourInTheYear);
+    }
+
+    void hourForEachArea(State& state, unsigned int numSpace)
+    {
+        for (uint clusterIndex = 0; clusterIndex != state.area->shortTermStorage.count();
+             ++clusterIndex)
+        {
+            // ST storage injection for the current cluster and this hour
+            // CashFlow[h] = (withdrawal - injection) * MRG. PRICE
+            pValuesForTheCurrentYear[numSpace][clusterIndex].hour[state.hourInTheYear]
+            = (state.hourlyResults->ShortTermStorage[state.hourInTheWeek].withdrawal[clusterIndex]
+            - state.hourlyResults->ShortTermStorage[state.hourInTheWeek].injection[clusterIndex])
```

Comment:
You may use a reference on `state.hourlyResults->ShortTermStorage[state.hourInTheWeek]` to avoid code duplication and make this more readable.